### PR TITLE
Al ticket24 cart resource edit

### LINKED ISSUE #15: Bug: Division by zero error on products
# CRITICAL ISSUE

When the client requests a single product, or all products, the status code is 500 and the response is always the following.

```
ZeroDivisionError at /products
division by zero
```

The most recent feature added by the team was adding an average rating key to each product. You might want to start there and see if there are any `try/catch` blocks you can implement.

### LINKED ISSUE #24: Bug: Cart resource has duplicate line items
When a client requests the `/profile/cart` resource, the following two keys are on the response object.

1. `lineitems`
1. `line_items`

The `line_items` key needs to be removed from the response.

### PR DESCRIPTION
**Pull request by alambrow**
Description of PR that completes issue here...

## Changes

- Altered doc descriptions and key-value pair initiation of "lineitems" in `/views/cart.py` and `views/profile.py`
- Altered `product.py` to handle cases with zero ratings when average_rating property is invoked on product class instance

## Requests / Responses

**Request**

GET `/profile/cart` (No body)

**Response**

HTTP/1.1 201 OK

```json
{
    "id": 2,
    "url": "http://localhost:8000/orders/2",
    "created_date": "2019-04-12",
    "payment_type": null,
    "customer": "http://localhost:8000/customers/7",
    "lineitems": [
        {
            "id": 4,
            "product": {
                "id": 52,
                "name": "900",
                "price": 1296.98,
                "number_sold": 0,
                "description": "1987 Saab",
                "quantity": 2,
                "created_date": "2019-03-19",
                "location": "Vratsa",
                "image_path": null,
                "average_rating": 0
            }
        },
        {
            "id": 5,
            "product": {
                "id": 33,
                "name": "Stratus",
                "price": 1199.91,
                "number_sold": 0,
                "description": "2001 Dodge",
                "quantity": 1,
                "created_date": "2019-04-06",
                "location": "Tianning",
                "image_path": null,
                "average_rating": 0
            }
        },
        {
            "id": 6,
            "product": {
                "id": 71,
                "name": "Sebring",
                "price": 1045.66,
                "number_sold": 0,
                "description": "1999 Chrysler",
                "quantity": 4,
                "created_date": "2019-05-18",
                "location": "Namibe",
                "image_path": null,
                "average_rating": 0
            }
        }
    ],
    "size": 3
}
```

## Testing

Description of how to test code...

- [ ] Pull down `al-ticket24-cart_resource_edit` locally
- [ ] Run migrations
- [ ] Seed database
- [ ] Run GET fetch call on `/profile/cart` in POSTMAN

## Related Issues

- Fixes #24
- Fixes #15

### CODE REVIEW

File: bangazon/urls.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 router = routers.DefaultRouter(trailing_slash=False)
 router.register(r'products', Products, 'product')
 router.register(r'productcategories', ProductCategories, 'productcategory')
-router.register(r'lineitems', LineItems, base_name='orderproduct')
+# router.register(r'lineitems', LineItems, base_name='orderproduct')
```
Comment (hannahhall) [inline bangazon/urls.py]: This line should not be commented out. To fix the bug from earlier remove the `basename=` from the arguments
```suggestion
router.register(r'lineitems', LineItems, 'orderproduct')
```

File: bangazonapi/models/product.py [hunk 2]
```diff
@@ -62,7 +62,10 @@ def average_rating(self):
         for rating in ratings:
             total_rating += rating.rating
 
-        avg = total_rating / len(ratings)
+        if len(ratings) != 0:
```
Comment (hannahhall) [inline bangazonapi/models/product.py]: This is covered in ticket #15 Can you update this to use a try/except block instead?

File: bangazonapi/views/profile.py [hunk 3]
```diff
@@ -177,15 +177,15 @@ def cart(self, request):
...
+                lineitems = OrderProduct.objects.filter(order=open_order)
+                lineitems = LineItemSerializer(
+                    lineitems, many=True, context={'request': request})
 
                 cart = {}
                 cart["order"] = OrderSerializer(open_order, many=False, context={
                                                 'request': request}).data
-                cart["order"]["line_items"] = line_items.data
-                cart["order"]["size"] = len(line_items.data)
+                cart["order"]["lineitems"] = lineitems.data
```
Comment (hannahhall) [inline bangazonapi/views/profile.py]: There's still `lineitems` and `line_items` returning the same thing. You'll want to remove one of those keys from the response. Either remove this line or check what is being returned in the `OrderSerializer`

**Review by hannahhall — CHANGES_REQUESTED** [review]

File: bangazon/urls.py (hunk 1 above)
Comment (alambrow) [reply bangazon/urls.py]: Uncommented and deleted `base_name=` from line 13 in latest commit

File: bangazonapi/models/product.py (hunk 2 above)
Comment (alambrow) [reply bangazonapi/models/product.py]: Updated the logic here to be `try / except ZeroDivisionError`

**Review by hannahhall — APPROVED** [review]
Looks good!

File: bangazonapi/views/profile.py (hunk 3 above)
Comment (alambrow) [reply bangazonapi/views/profile.py]: Removed line 187 and preserved the return in OrderSerializer, which itself handles generation of "lineitems" key-value pair.

**Pull request merged**